patches/server/0670-Missing-Entity-Behavior-API.patch

```diff
@@ -24,6 +24,66 @@ index d13f3460644f635ded96bf92ddf9ecf8984c8e47..bd048cc30046f19f9eee89c6ba45d081
          this.setFlag(2, nearTarget);
      }
  
+diff --git a/src/main/java/net/minecraft/world/entity/animal/frog/Tadpole.java b/src/main/java/net/minecraft/world/entity/animal/frog/Tadpole.java
+index 3dd4e6d622a6daafa00ae971edd88a147e34beef..1740f7caa5f39b45466598b800403262a6e89744 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/frog/Tadpole.java
++++ b/src/main/java/net/minecraft/world/entity/animal/frog/Tadpole.java
+@@ -44,6 +44,7 @@ public class Tadpole extends AbstractFish {
+     public int age;
+     protected static final ImmutableList<SensorType<? extends Sensor<? super Tadpole>>> SENSOR_TYPES = ImmutableList.of(SensorType.NEAREST_LIVING_ENTITIES, SensorType.NEAREST_PLAYERS, SensorType.HURT_BY, SensorType.FROG_TEMPTATIONS);
+     protected static final ImmutableList<MemoryModuleType<?>> MEMORY_TYPES = ImmutableList.of(MemoryModuleType.LOOK_TARGET, MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES, MemoryModuleType.WALK_TARGET, MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE, MemoryModuleType.PATH, MemoryModuleType.NEAREST_VISIBLE_ADULT, MemoryModuleType.TEMPTATION_COOLDOWN_TICKS, MemoryModuleType.IS_TEMPTED, MemoryModuleType.TEMPTING_PLAYER, MemoryModuleType.BREED_TARGET, MemoryModuleType.IS_PANICKING);
++    public boolean ageLocked; // Paper
+ 
+     public Tadpole(EntityType<? extends AbstractFish> type, Level world) {
+         super(type, world);
+@@ -94,7 +95,7 @@ public class Tadpole extends AbstractFish {
+     @Override
+     public void aiStep() {
+         super.aiStep();
+-        if (!this.level.isClientSide) {
++        if (!this.level.isClientSide && !this.ageLocked) { // Paper
+             this.setAge(this.age + 1);
+         }
+ 
+@@ -104,12 +105,14 @@ public class Tadpole extends AbstractFish {
+     public void addAdditionalSaveData(CompoundTag nbt) {
+         super.addAdditionalSaveData(nbt);
+         nbt.putInt("Age", this.age);
++        nbt.putBoolean("AgeLocked", this.ageLocked); // Paper
+     }
+ 
+     @Override
+     public void readAdditionalSaveData(CompoundTag nbt) {
+         super.readAdditionalSaveData(nbt);
+         this.setAge(nbt.getInt("Age"));
++        this.ageLocked = nbt.getBoolean("AgeLocked"); // Paper
+     }
+ 
+     @Nullable
+@@ -162,6 +165,7 @@ public class Tadpole extends AbstractFish {
+         CompoundTag nbttagcompound = stack.getOrCreateTag();
+ 
+         nbttagcompound.putInt("Age", this.getAge());
++        nbttagcompound.putBoolean("AgeLocked", this.ageLocked); // Paper
+     }
+ 
+     @Override
+@@ -171,6 +175,7 @@ public class Tadpole extends AbstractFish {
+             this.setAge(nbt.getInt("Age"));
+         }
+ 
++        this.ageLocked = nbt.contains("AgeLocked") && nbt.getBoolean("AgeLocked"); // Paper
```

Comment:
The getBoolean method will already returns false in case the key is not found so no need to check using the contains method before